File: packages/strategy-runtime/src/strategy.ts
import type { Bar, Tick, Order } from '@quant/common';
import type { StrategyMeta } from './meta.js';
import type { StrategyContext } from './context.js';
import type { StrategyResult } from './result.js';
import type { StrategyState } from './types.js';

export interface Strategy {
  readonly meta: StrategyMeta;
  readonly state: StrategyState;
  init(context: StrategyContext): void;
  onBar(bar: Bar, context: StrategyContext): void;
  onTick?(tick: Tick, context: StrategyContext): void;
  onOrder?(order: Order, context: StrategyContext): void;
  finish(): StrategyResult;
}

File: packages/strategy-runtime/quantforge_strategy/strategy.py
"""策略抽象基类"""

from __future__ import annotations

from abc import ABC, abstractmethod

from .context import StrategyContext
from .market import Bar, Tick
from .meta import StrategyMeta
from .order import Order
from .result import StrategyResult
from .types import StrategyState


class Strategy(ABC):
    @property
    @abstractmethod
    def meta(self) -> StrategyMeta: ...

    @property
    @abstractmethod
    def state(self) -> StrategyState: ...

    @abstractmethod
    def init(self, context: StrategyContext) -> None: ...

    @abstractmethod
    def on_bar(self, bar: Bar, context: StrategyContext) -> None: ...

    def on_tick(self, tick: Tick, context: StrategyContext) -> None:
        pass

    def on_order(self, order: Order, context: StrategyContext) -> None:
        pass

    @abstractmethod
    def finish(self) -> StrategyResult: ...
